02-Architecture Constraints.md

```diff
@@ -1,3 +1,17 @@
 # Architecture Constraints
 
+An architecture constraint is any requirement that constrains software developers in their freedom of design and implementation decisions or decision about the development process. These constraints sometimes go beyond individual systems and are valid for whole organizations.
 
+## Design and Development constraints
+
+Technical constraints that apply to the design and development of the system.
+
+## [Lua](https://en.wikipedia.org/wiki/Wikipedia:Lua)
+
+Lua is a programming language that is available via the [Scribunto MediaWiki extension](https://www.mediawiki.org/wiki/Extension:Scribunto) on the English Wikipedia. Lua code can be embedded into [wiki templates](https://en.wikipedia.org/wiki/Help:A_quick_guide_to_templates). It's often used in templates for [Infoboxes](https://en.wikipedia.org/wiki/Help:Infobox) on Wikipedias.
+
+The WikibaseClient extension provides a Lua Scribunto interface that implements functions to access data from the Wikibase repository. Lua modules and wiki templates can invoke these functions.
```

Comment:
I don't like this sentence in terms of constraints.
I think for both this and wikitext the constraint that we want to try to represent is that we want wikidata data to be available in wikitext and in LUA.